### ISSUE TITLE
repository structure

### ISSUE THREAD

**Issue by zbpetersbuf** [ubsuny/CP1-24-HW7#10]
@ubsuny/cp1-24 what structure of the repository do we all want, do we want to use the same files calculus.py, test_calculus.py or does every one want to use different files? Using different files means you need to be more careful when calling functions but does make merging easier.

**Comment by laserlab** [comment]
If you do not get timely feedback you might just have to decide on something and then if it is not ideal change it during the homework.

**Comment by zbpetersbuf** [comment]
Yes that was my plan, so unless anyone else speaks up by about 8 pm we will be sharing files, (but there is one for, there will need to be two against to change this)

**Comment by zbpetersbuf** [comment]
Alright, I will now make the calculus.py and  test_calculus.py and everyone will share files, the same way how the last hw went

**Comment by ojha-aditya** [comment]
An additional suggestion that I had was to pre-decide reviewers alongside having shared files. That way we can speed up the merging process causing minimum merge conflicts.

**Comment by ojha-aditya** [comment]
As most tasks have 2 members working on them, we can maybe review the PRs in that order, for example me and @SchrodingersStruggle work on the Trapezoid algorithm so can review each others PR. Should we do this?

**Comment by ojha-aditya** [comment]
Also should we include a documentation file, explaining the algorithms in the repository structure?

**Comment by zbpetersbuf** [comment]
I mean this is a good idea but it assumes that people turn in their hw on time and when then make a pull request they check back to see if its actually good to merge, but looking back at previous hw's this doesn't tend to happen, if we want to do this we can, only I feel like this will sure cause issues quickly

**Comment by ojha-aditya** [comment]
> I mean this is a good idea but it assumes that people turn in their hw on time and when they make a pull request they check back to see if its actually good to merge, but looking back at previous hw's this doesn't tend to happen, if we want to do this we can, only I feel like this will surely cause issues quickly

That is true. I guess lets try this but not very rigidly then.

**Comment by Priyanka-43** [comment]
Hey, I have still confusion, how to make second pull request. For Example, I am working for simpson rule, and created calculus.py and test_calculus.py, I have create one pull request. But now i will do one more task that is implementation in same. then how i can make new pull request?

**Comment by HeshamElsaman** [comment]
> Hey, I have still confusion, how to make second pull request. For Example, I am working for simpson rule, and created calculus.py and test_calculus.py, I have create one pull request. But now i will do one more task that is implementation in same. then how i can make new pull request?

@Priyanka-43 
If I get it right, if you mean that you already made a PR and it's not accepted yet, you can make another one from a different branch.

**Comment by Priyanka-43** [comment]
Yeah, i want to ask that. Can i make different branch and then create second pull request?

**Comment by HeshamElsaman** [comment]
> Yeah, i want to ask that. Can i make different branch and then create second pull request?

I guess so. You'll just need to deal with the conflicts with the different branches when merging, that's to my understanding.

**Comment by zbpetersbuf** [comment]
Ya you should be able to do that with a different branch

**Comment by ojha-aditya** [comment]
> Ya you should be able to do that with a different branch

I think we should try to review the existing PRs quickly because this would be really confusing if everyone pushes two PRs from different branches. Will have too many merge conflicts.

**Comment by Priyanka-43** [comment]
Yes, that is true.

**Comment by zbpetersbuf** [comment]
ya do you want me to assign people? or assign yourself? I was going to let people assign themselves to reviews, since that's what I have been doing for the most part but now im not sure if that's how everyone been functioning

**Comment by kylemasc917** [comment]
@zbpetersbuf Since we are getting closer to the deadline you should start assigning people who have been good reviewers already during this assignment to pick up the slack and leave another issue open for people to volunteer.

**Comment by kylemasc917** [comment]
I myself am willing to help review if needed.

**Comment by kylemasc917** [comment]
Also, most merge conflicts can be pretty easily resolved so people shouldn't be too worried about multiple pull requests from different branches.

**Comment by kylemasc917** [comment]
Other than that, it's just about giving consistent reminders to those that have taken some time to do certain tasks.
